scripts/compile.js

```diff
@@ -0,0 +1,334 @@
+ * Compile
+ **********************************************************************/
+
+async function compile(page) {
+    console.time(page.save_as);
+    console.log(`Compiling ${page.save_as}`.green);
+    const config = getConfig();
+    const languages = config.languages.filter(lang => shouldCompile(page.excludes, lang));
+    const builder = await createBuilder();
+
+    const layouts = { };
+    await Promise.all(
+        config.sections.map(async section => {
+            layouts[section] = {
+                file: Path.join(config.root_path, `src/templates/${section}/_layout/layout.vash`),
+            }
+            layouts[section].template = await builder.build_template_for(layouts[section].file);
+        })
+    );
+
+    const input = {
+        file: Path.join(config.root_path, `src/templates/${page.tpl_path}.vash`),
+        template: null,
+    };
+    input.template = await builder.build_template_for(input.file);
+
+    const tasks = languages.map(async lang => {
+
+        const model = {
+            website_name : 'Binary.com',
+            title        : page.title,
+            layout       : page.layout,
+            language     : lang.toUpperCase(),
+            root_url     : config.root_url,
+            only_ja      : page.only_ja,
+            current_path : page.save_as,
+            current_route: page.current_route,
+            affiliate_email : 'affiliates@binary.com',
+            japan_docs_url  : 'https://japan-docs.binary.com',
+            is_pjax_request : true,
+        }
+
+        const result_input = builder.run_template({template: input.template, model: model});
+
+        if (page.layout) {
+            const layout = layouts[page.tpl_path.split('/')[0]];
+            const result_pjax = builder.run_template({template: layout.template, model: model})
+
+            await common.writeFile( // pjax layout
+                Path.join(config.dist_path, `${lang}/pjax/${page.save_as}.html`),
+                result_pjax.replace(CONTENT_PLACEHOLDER, result_input),
+                'utf8'
+            );
+
+            model.is_pjax_request = false;
+            const result_normal = builder.run_template({template: layout.template, model: model})
+
+            await common.writeFile( // normal layout
+                Path.join(config.dist_path, `${lang}/${page.save_as}.html`),
+                result_normal.replace(CONTENT_PLACEHOLDER, result_input),
+                'utf8'
+            );
+        } else {
+            await common.writeFile( // landing pages
+                Path.join(config.dist_path, `${lang}/${page.save_as}.html`),
+                result_input,
+                'utf8'
+            );
+        }
+    });
+    await Promise.all(tasks);
+    console.timeEnd(page.save_as);
+}
+
+createDirectories();
+(async () => {
+    const config = getConfig();
+    try {
+        await compile(common.pages.find(p => p.save_as === (program.path || 'home')));
```

Comment:
Would be more convenient to allow using regular expressions like the current one:
```js
await compile(common.pages.filter(p => new RegExp(program.path, 'i').test(p.save_as)));
```